mne/viz/_3d.py

```diff
@@ -1919,22 +1897,31 @@ def _plot_stc(stc, subject, surface, hemi, colormap, time_label,
             "verbose": False
         }
         if using_mayavi:
+            kwargs["array"] = getattr(stc, hemi + '_data')
+            kwargs["vertices"] = stc.vertices[0 if hemi == 'lh' else 1]
             kwargs["min"] = scale_pts[0]
             kwargs["mid"] = scale_pts[1]
             kwargs["max"] = scale_pts[2]
         else:  # pyvista
+            kwargs["array"] = stc
             kwargs["fmin"] = scale_pts[0]
             kwargs["fmid"] = scale_pts[1]
             kwargs["fmax"] = scale_pts[2]
             kwargs["clim"] = clim
-            kwargs["volume_options"] = volume_options
-            kwargs["src"] = src_vol
+            kwargs["src"] = src
         kwargs.update({} if add_data_kwargs is None else add_data_kwargs)
+        if volume and src.kind == 'volume':
+            continue
         with warnings.catch_warnings(record=True):  # traits warnings
             brain.add_data(**kwargs)
         brain.scale_data_colormap(fmin=scale_pts[0], fmid=scale_pts[1],
```

Comment:
Shouldn't this also be outside the loop and probably after the new `volume` lines?

Reply:
This skips the case where we don't want to add surface when there is none similarly to https://github.com/mne-tools/mne-python/blob/2b4605fbf50d1d03edc34ad013b5b4577f948b81/mne/viz/_3d.py#L1901-L1902

Reply:
Related to https://github.com/mne-tools/mne-python/pull/8547#discussion_r527768391 and https://github.com/mne-tools/mne-python/pull/8547#discussion_r527769357:

> someone calls add_data but no data gets added?